src/qutip_cupy/dense_functions.py

```diff
@@ -29,3 +29,36 @@ def trace_cupydense(cp_arr):
     # @TODO: whnen qutip allows it we should remove this call to item()
     # as it takes a time penalty commmunicating data from GPU to CPU.
     return cp.trace(cp_arr._cp).item()
+
+
+hermdiff_kernel = cp.RawKernel(
+    r"""
+    #include <cupy/complex.cuh>
+    extern "C" __global__
+    void hermdiff(const complex<double>* x1,const int size,const double tol, bool* y) {
+        for (unsigned int tidx = blockDim.x * blockIdx.x + threadIdx.x; tidx < size;
+                                                    tidx += gridDim.x * blockDim.x) {
+        for (unsigned int tidy = blockDim.y * blockIdx.y + threadIdx.y;
+                                        tidy < size; tidy += gridDim.y * blockDim.y) {
+
+            y[tidx+size*tidy] = norm(x1[tidx*size+tidy]
+                                    - conj(x1[tidy*size+tidx])) < tol;
+        };
+        };
+    }""",
+    "hermdiff",
+)
+
+
+def isherm_cupydense(cp_arr, tol):
+    if cp_arr.shape[0] != cp_arr.shape[1]:
+        return False
+    size = cp_arr.shape[0]
+    diff = cp.empty((size, size), dtype=cp.bool_)
```

Comment:
Another optimization: Only do this for upper or lower triangular matrix. Currently we're doing twice of the necessary work.

Reply:
Good point, I'llsee if  I cand do it while not increasing thread divergence too much.